sdks/python/apache_beam/io/gcp/bigquery.py

```diff
@@ -644,6 +644,8 @@ def estimate_size(self):
           self.table_reference.tableId)
       return int(table.numBytes)
     else:
+      if isinstance(self.query, ValueProvider):
+        self.query = self.query.get()
```

Comment:
I would probably do this without changing the class variable.

I'd do a local variable: `query = self.query.get()`, and do it on every instance where `self.query` is accessed.

The reason is that the Source objects may be recreated in different workers, and the  `self.query` variable would contain the original ValueProvider in all of those.